File: Add Digits.py
# Solution 1
# class Solution:
    # # @param {integer} num
    # # @return {integer}
    # def addDigits(self, num):
        # if num < 10:
            # return num
        # res = num % 10
        # while True:
            # res += num % 10
            # is_digit = num / 10
        # return addDigits(res)

# Solution 2
class Solution:
    # @param {integer} num
    # @return {integer}
    def addDigits(self, num):
        if num % 9 :
            return num % 9
        elif num:
            return 9
        else:
            return 0

print Solution().addDigits(38)

File: javascript/Add Digits.js
/**
 * @param {number} num
 * @return {number}
 */

var addDigits = function(num) {
    // debug
    console.log("num is " + num);
    res = 0
    while (Math.floor(num / 10) > 0){
        res += num % 10;
        num = Math.floor(num / 10);
    }
    res += Math.floor(num);

    if (Math.floor(res / 10) > 0){
        addDigits(res);
    }
    return res;
};

console.log(addDigits(38));

